src/tests/ftest/util/apricot/apricot/test.py

```diff
@@ -639,7 +639,8 @@ def __init__(self, *args, **kwargs):
         self.job_manager = None
         # whether engines ULT stacks have been already dumped
         self.dumped_engines_stacks = False
-        self.label_generator = LabelGenerator()
+        self.pool_label_generator = LabelGenerator()
+        self.cont_label_generator = LabelGenerator()
```

Comment:
Maybe we should use a single label generator object that could maintain unique counts per base label, e.g. A1, A2, B1, A3, B2, A4, etc.

Reply:
Yes, agree. Thanks

Reply:
Addressed in the latest merge commit